modules/fileout_netcdf/FONcArrayStructureField.cc

```diff
@@ -299,6 +400,121 @@ void FONcArrayStructureField::obtain_scalar_data(char *data_buf_ptr, BaseType *b
     }
 }
 
+// Note: TODO: if the string is equal-size, we can optimize as write_equal_length_string_array() in FONcArray.cc.
+//             However, for netCDF-4, the better way is to map DAP4 string to netCDF-4 string. 
+
+void FONcArrayStructureField::write_str(int ncid){
+
+    size_t d_ndims = struct_dims.size();
+    vector<size_t> var_count(d_ndims);
+    vector<size_t> var_start(d_ndims);
+
+    int dim = 0;
+    for (dim = 0; dim < d_ndims; dim++) {
+        // the count for each of the dimensions will always be 1 except
+        // for the string length dimension.
+        // The size of the last dimension (var_count[d_ndims-1]) is set
+        // separately for each element below. jhrg 10/3/22
+        var_count[dim] = 1;
+
+        // the start for each of the dimensions will start at 0. We will
+        // bump this up in the while loop below
+        var_start[dim] = 0;
+    }
+
+    // Obtain the compound_buf; this is for gathering the data for individual fields.
+    vector<BaseType*> compound_buf = d_a->get_compound_buf();
+
+    for (unsigned i= 0; i<d_a->length_ll();i++) {
```

Comment:
This could be replaced with 'for (auto &cb : compound_buf) { ...', as above.

Reply:
This cannot be replaced by range-get since the index i is used in this for loop.